src/pybamm/solvers/summary_variable.py

```diff
@@ -81,25 +84,27 @@ def all_variables(self) -> list[str]:
         Return names of all possible summary variables, including eSOH variables
          if appropriate.
         """
-        try:
+        if self._all_variables is not None:
```

Comment:
`None` is a value. This block is checking if the variable even exists; hence, the new logic will not work. Something like this should work, but please check -
```suggestion
        if getattr(self, "_all_variables", None) is not None:
```

Reply:
mypy still thinks none can be returned so I think it should be something like this:
```python
all_vars = getattr(self, "_all_variables", None)
if all_vars is not None:   
  ```